include/psen_scan/start_request.h

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2020 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#ifndef PSEN_SCAN_START_REQUEST_H
+#define PSEN_SCAN_START_REQUEST_H
+
+#include <arpa/inet.h>
+#include <cstdint>
+#include <endian.h>
+#include <string>
+#include <vector>
+
+#include <psen_scan/scanner_configuration.h>
+#include <psen_scan/crc.h>
+
+#pragma pack(push, 1)
+
+namespace psen_scan
+{
+
+/**
+ * @brief Frame containing all necessary fields for a Start Request.
+ *
+ * Unless otherwise indicated the byte order is little endian.
+ *
+ */
+typedef struct StartRequest
+{
+public:
+  uint32_t crc_;                  /**< A CRC32 of all the following fields. */
+  uint32_t seq_number_;
+  uint64_t const RESERVED_ { 0 };       /**< Use all zeros */
+  uint32_t const OPCODE_ { htole32(0x35) };         /**< Constant 0x35. */
+  uint32_t target_ip_;            /**< Byte order: big endian */
+  uint16_t target_udp_port_;  /**< Byte order: big endian */
+
+  /**< The following 'enable' fields are a 1-byte mask each.
+   * Only the last 4 bits (little endian) are used, each of which represents a device.
+   * For example, (1000) only enables the Master device, while (1010) enables both the Master
+   * and the second Slave device.
+   */
+  uint8_t device_enabled_;
+  uint8_t intensity_enabled_;
+  uint8_t point_in_safety_enabled_;
+  uint8_t active_zone_set_enabled_;
+  uint8_t io_pin_enabled_;
+  uint8_t scan_counter_enabled_;
+  uint8_t speed_encoder_enabled_;    /**< 0000000bin disabled, 00001111bin enabled.*/
+  uint8_t diagnostics_enabled_;
+
+  /**< The following 'angle' and 'resolution' fields are all in tenth of degrees */
+  uint16_t master_start_angle_;
+  uint16_t master_end_angle_;
+  uint16_t master_resolution_;
+  uint16_t slave1_start_angle_;
```

Comment:
I suggest using one array here for the different `slaveK_start_angle_`, `slaveK_end_angle_` and `slaveK_resolution_` variables. This way it is easier to extend the slave. A getter function could then simplify the access to the different angle variables of the slaves.

Reply:
done